crates/ruff_linter/src/checkers/ast/mod.rs

```diff
@@ -174,11 +178,42 @@ impl ExpectedDocstringKind {
     }
 }
 
+struct ParsedAnnotationsCache<'a> {
```

Comment:
Nit: I would move this type to the end of the file. It seems less important than `Checker` (which is the main thing in this file)